crates/dslab-iaas/readme.md

```diff
@@ -0,0 +1,87 @@
+# DSLab IaaS
+
+## Overview
+
+A library for cloud simulations creation.
+
+It can be used by cloud computing academic studies to implement user specific simulations. The simulator contains a set of standard components which allow developers to focus on specific system design issues to be investigated, without concerning the low-level details of simulation. At the same time, any component can be easily extended or replaced by the user to meet their behavior requirements. As the result, the library supports simple, easy-to-understand simulations and also grants the access to low-level details.
+
+The library enables modelling of different cloud infrastructure items such as physical machines (or [HostManager](https://github.com/osukhoroslov/dslab/blob/main/crates/dslab-iaas/src/core/host_manager.rs)), virtual machines ([VirtualMachine](https://github.com/osukhoroslov/dslab/blob/main/crates/dslab-iaas/src/core/vm.rs)) and other components of resource allocation and management. The simulation processes a chain of discrete events and their handlers brought from [dslab-core](https://github.com/osukhoroslov/dslab/tree/main/crates/dslab-core) crate. Any component implements the handler interface and use events to interact with others.
+
+The simulation is managed via [CloudSimulation](https://github.com/osukhoroslov/dslab/blob/main/crates/dslab-iaas/src/simulation.rs#L25), which provides an interface to spawn new physical machines, submit virtual machines and migrate them, add new user defined components etc. As long as simulation consists of events ordered by timestamp, it is possible to run simulation for a few steps or time period. 
```

Comment:
The simulation is configured and managed via [CloudSimulation](src/simulation.rs). It encapsulates all simulations components and provides convenient access to them for the user. It provides an interface to add new physical machines, submit virtual machines and migrate them, add new user defined components etc. As long as the simulation has some unprocessed events, it is possible to run simulation for a few steps or time period by processing events in the order of their timestamps.